### ISSUE TITLE
Helm release panic when changing values from non-output to output type

### ISSUE THREAD

**Issue by gitfool** [pulumi/pulumi-kubernetes#1933]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

Using the following versions:
* `Pulumi` `3.26.1`
* `Pulumi.Kubernetes` `3.16.0`

I hit the following panic when using the dotnet sdks:
```
Diagnostics:
  kubernetes:helm.sh/v3:Release (fluent-bit):
    error: transport is closing

  pulumi:pulumi:Stack (k8s-ci):
    panic: fatal: An assertion has failed
    goroutine 379 [running]:
    github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.24.1/go/common/util/contract/failfast.go:23
    github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.Assert(...)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.24.1/go/common/util/contract/assert.go:26
    github.com/pulumi/pulumi-kubernetes/provider/v3/pkg/provider.(*kubeProvider).Check(0xc000600d80, {0x2b016a8, 0xc002587bc0}, 0xc00f0d07d0)
        /home/runner/work/pulumi-kubernetes/pulumi-kubernetes/provider/pkg/provider/provider.go:1239 +0x1579
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Check_Handler.func1({0x2b016a8, 0xc002587bc0}, {0x2639a80, 0xc00f0d07d0})
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.24.1/proto/go/provider.pb.go:2593 +0x78
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x2b016a8, 0xc002586ed0}, {0x2639a80, 0xc00f0d07d0}, 0xc003dc3840, 0xc00bffc6c0)
        /home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x425
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Check_Handler({0x26de380, 0xc000600d80}, {0x2b016a8, 0xc002586ed0}, 0xc003ea5b00, 0xc00077fac0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.24.1/proto/go/provider.pb.go:2595 +0x138
    google.golang.org/grpc.(*Server).processUnaryRPC(0xc0007b2c40, {0x2b426e8, 0xc000685ba0}, 0xc00b5af9e0, 0xc000884840, 0x3f24a50, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.43.0/server.go:1282 +0xccf
    google.golang.org/grpc.(*Server).handleStream(0xc0007b2c40, {0x2b426e8, 0xc000685ba0}, 0xc00b5af9e0, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.43.0/server.go:1616 +0xa2a
    google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.43.0/server.go:921 +0x98
    created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.43.0/server.go:919 +0x294
```

### Steps to reproduce

1. Given an existing k8s stack with the following "helm release" for [fluent bit](https://github.com/fluent/helm-charts/tree/main/charts/fluent-bit):
```c#
var fluentBitValues =
    new Dictionary<string, object>
    {
        ["image"] = new
        {
            repository = K8sConfig.FluentBitImageRepository,
            tag = K8sConfig.FluentBitImageTag,
            pullPolicy = "IfNotPresent"
        },
        ["priorityClassName"] = "system-cluster-critical",
        ["resources"] = new
        {
            requests = new { cpu = "50m", memory = "50Mi" },
            limits = new { memory = "100Mi" }
        },
        ["serviceMonitor"] = new { enabled = true },
        ["prometheusRule"] = new
        {
            enabled = true,
            rules = new[]
            {
                new
                {
                    alert = "FluentBitNoOutputBytesProcessed",
                    expr = "rate(fluentbit_output_proc_bytes_total[5m]) == 0",
                    annotations = new
                    {
                        description = "Fluent Bit instance {{ $labels.instance }} output plugin {{ $labels.name }} has not processed any bytes for at least 15 minutes.",
                        summary = "No output bytes processed"
                    },
                    @for = "15m",
                    labels = new { severity = "critical" }
                }
            }
        },
        ["dashboards"] = new { enabled = true },
        ["config"] = new
        {
            service = ReadResource("FluentBitService.ini"),
            inputs = ReadResource("FluentBitInputs.ini"),
            filters = ReadResource("FluentBitFilters.ini"),
            outputs = ReadResource("FluentBitOutputs.ini")
        },
        ["luaScripts"] = new Dictionary<string, string>
        {
            ["filters.lua"] = ReadResource("FluentBitFilters.lua")
        },
        ["extraVolumeMounts"] = new[] // workaround scripts directory not found
        {
            new
            {
                name = "luascripts",
                subPath = "filters.lua",
                mountPath = "/fluent-bit/etc/filters.lua"
            }
        },
        ["tolerations"] = new[]
        {
            new { effect = "NoExecute", @operator = "Exists" },
            new { effect = "NoSchedule", @operator = "Exists" }
        }
    }.ToDictionary(); // workaround https://github.com/pulumi/pulumi/issues/8013

new Release("fluent-bit",
    new ReleaseArgs
    {
        Namespace = "kube-system",
        Name = "fluent-bit",
        RepositoryOpts = new RepositoryOptsArgs { Repo = "https://fluent.github.io/helm-charts" },
        Chart = "fluent-bit",
        Version = K8sConfig.FluentBitChartVersion,
        Values = fluentBitValues,
        Atomic = true
    },
    new CustomResourceOptions { DependsOn = kubePrometheusStackCrds.Ready(), Provider = k8sProvider });
```
2. Change the code to use [iam roles for service accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) with eks:
```c#
var fluentBitRole = new RoleX($"{k8sPrefix}-fluent-bit",
    new RoleXArgs
    {
        AssumeRolePolicy = IamHelpers.AssumeRoleForServiceAccount(oidcArn, oidcUrl, "kube-system", "fluent-bit", awsProvider),
        InlinePolicies = { ["policy"] = ReadResource("FluentBitPolicy.json") }
    },
    new ComponentResourceOptions { Provider = awsProvider });

var fluentBitValues = fluentBitRole.Arn.Apply(roleArn =>
    new Dictionary<string, object>
    {
        ["image"] = new
        {
            repository = K8sConfig.FluentBitImageRepository,
            tag = K8sConfig.FluentBitImageTag,
            pullPolicy = "IfNotPresent"
        },
        ["priorityClassName"] = "system-cluster-critical",
        ["resources"] = new
        {
            requests = new { cpu = "50m", memory = "50Mi" },
            limits = new { memory = "100Mi" }
        },
        ["serviceMonitor"] = new { enabled = true },
        ["prometheusRule"] = new
        {
            enabled = true,
            rules = new[]
            {
                new
                {
                    alert = "FluentBitNoOutputBytesProcessed",
                    expr = "rate(fluentbit_output_proc_bytes_total[5m]) == 0",
                    annotations = new
                    {
                        description = "Fluent Bit instance {{ $labels.instance }} output plugin {{ $labels.name }} has not processed any bytes for at least 15 minutes.",
                        summary = "No output bytes processed"
                    },
                    @for = "15m",
                    labels = new { severity = "critical" }
                }
            }
        },
        ["dashboards"] = new { enabled = true },
        ["config"] = new
        {
            service = ReadResource("FluentBitService.ini"),
            inputs = ReadResource("FluentBitInputs.ini"),
            filters = ReadResource("FluentBitFilters.ini"),
            outputs = ReadResource("FluentBitOutputs.ini")
        },
        ["luaScripts"] = new Dictionary<string, string>
        {
            ["filters.lua"] = ReadResource("FluentBitFilters.lua")
        },
        ["extraVolumeMounts"] = new[] // workaround scripts directory not found
        {
            new
            {
                name = "luascripts",
                subPath = "filters.lua",
                mountPath = "/fluent-bit/etc/filters.lua"
            }
        },
        ["serviceAccount"] = new { annotations = new Dictionary<string, string> { ["eks.amazonaws.com/role-arn"] = roleArn } },
        ["tolerations"] = new[]
        {
            new { effect = "NoExecute", @operator = "Exists" },
            new { effect = "NoSchedule", @operator = "Exists" }
        }
    }.ToDictionary()); // workaround https://github.com/pulumi/pulumi/issues/8013

new Release("fluent-bit",
    new ReleaseArgs
    {
        Namespace = "kube-system",
        Name = "fluent-bit",
        RepositoryOpts = new RepositoryOptsArgs { Repo = "https://fluent.github.io/helm-charts" },
        Chart = "fluent-bit",
        Version = K8sConfig.FluentBitChartVersion,
        Values = fluentBitValues,
        Atomic = true
    },
    new CustomResourceOptions { DependsOn = kubePrometheusStackCrds.Ready(), Provider = k8sProvider });
```

Expected: 🚀 
Actual: 💥 

As a workaround I commented out the code and ran it to delete the helm release resource, then I uncommented out the code and it ran without any problems. 🤔

**Comment by viveklak** [comment]
We have hit this before in https://github.com/pulumi/pulumi-kubernetes/issues/1925 as well.

**Comment by RazcoDev** [comment]
I'm posting my repro code as @viveklak requested here : https://github.com/pulumi/pulumi-kubernetes/issues/1925#issuecomment-1086132906

```typescript
import * as k8s from "@pulumi/kubernetes";
import * as pulumi from "@pulumi/pulumi";
import {FileAsset} from "@pulumi/pulumi/asset";
import * as aws from "@pulumi/aws";

export function installHelmBootstrapChart(stackName: string, kubeProvider: k8s.Provider, certificateArn: pulumi.Output<string>, externalDnsRoleArn: pulumi.Output<string>, externalSecretsRoleArn: pulumi.Output<string>, awsRegion: aws.Region) {
    const namespaceName = "eks-bootstrap"
    const releaseValue = {
        "external-dns": {
            "aws": {
                "roleArn": externalDnsRoleArn  // This is an Output<string> 
            }
        },
    }
    const releaseArgs: k8s.helm.v3.ReleaseArgs = {
        atomic: true,
        name: "eks-bootstrap",
        chart: "./charts/eks-bootstrap-chart",
        dependencyUpdate: true,
        namespace: namespaceName,
        values: releaseValue,
        valueYamlFiles: [new FileAsset(`./charts/eks-bootstrap-chart/values-${stackName}.yaml`)],
        forceUpdate: true,
        createNamespace: true
    }
    const release: k8s.helm.v3.Release = new k8s.helm.v3.Release("eks-bootstrap", releaseArgs, {
        provider: kubeProvider,
        dependsOn: [kubeProvider]
    })
```

(Posted also on : https://github.com/pulumi/pulumi-kubernetes/issues/1942)

**Comment by viveklak** [comment]
Thanks for the repro @RazcoDev. I was able to come up with a simple representation of this. A fix is in #1953.
